src/openms/include/OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/IsotopePatternSolver.h

```diff
@@ -0,0 +1,81 @@
+
+#ifndef OPENMS_CHEMISTRY_ISOTOPEDISTRIBUTION_ISOTOPEPATTERNSOLVER_H
+#define OPENMS_CHEMISTRY_ISOTOPEDISTRIBUTION_ISOTOPEPATTERNSOLVER_H
+
+
+
+#include <OpenMS/CHEMISTRY/Element.h>
+#include <OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/Container.h>
+#include <OpenMS/CONCEPT/Types.h>
+
+#include <deque>
+#include <algorithm>
+
+namespace OpenMS
+{
+  class OPENMS_DLLAPI IsotopePatternSolver : public IsotopeDistribution
+  {
+ public:
+    IsotopePatternSolver();
+    IsotopePatternSolver(const IsotopeDistribution&);
+    virtual void run() = 0;
+    void dumpIDToFile(String file);
+    void setEmpiricalFormula(const EmpiricalFormula&);
```

Comment:
just out of interest. Is there a reason why the formula needs to be stored? Would an interface that returns an IsotopeDistrubtion from an EF also work?
e.g.: virtual IsotopeDistribution run(const EmpiricalFormula& ef) ?